### PR TITLE
[B+C] Add special event for breeding. Adds BUKKIT-5357

### PR DESCRIPTION
**Pull request by ltcyx**
The Issue:

It could be interesting to have a special even for when a new LivingEntity is spawned because of breeding. It should contain references to both parents so that, for example, wool colour from both can be used in detemining the colour for the offspring.

Justification for this PR:

This adds CreatureBreedSpawnEvent, which extends CreatureSpawnEvent and is fired when a new creature is spawned because of breeding. This new event contains references to both parents, which can then be examined to alter the offspring or modified themselves.

PR Breakdown:

This adds CreatureBreedSpawnEvent.

Testing Results and Materials:

A custom CraftBukkit server was built with this, and a plugin was created with the patched Bukkit lib. This plugin captures this new event and, if the new entity is a sheep, it colours the wool according to the colour of the parents, blending it sort of like you get when mixing dyes in a workbench

Relevant PR(s):

[B+C]-1330 - https://github.com/Bukkit/CraftBukkit/pull/1330 - adds CraftBukkit support for the event

JIRA Ticket:

BUKKIT-5357 https://bukkit.atlassian.net/browse/BUKKIT-5357

### CODE REVIEW

File: src/main/java/org/bukkit/event/entity/CreatureBreedSpawnEvent.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Called when a creature spawns because of breeding.
+ */
+public class CreatureBreedSpawnEvent extends CreatureSpawnEvent {
+	private LivingEntity parent1;
+	private LivingEntity parent2;
+	
+	public CreatureBreedSpawnEvent(LivingEntity spawnee, LivingEntity parent1, LivingEntity parent2) {
+		super(spawnee, SpawnReason.BREEDING);
+		// TODO Auto-generated constructor stub
```
Comment (natemort) [inline src/main/java/org/bukkit/event/entity/CreatureBreedSpawnEvent.java]: Umm....

Comment (ltcyx) [inline src/main/java/org/bukkit/event/entity/CreatureBreedSpawnEvent.java]: bloody templates. I've became blind to those comments. Should I make another PR?

Comment (natemort) [inline src/main/java/org/bukkit/event/entity/CreatureBreedSpawnEvent.java]: You've pushed commits to fix the issue, so it's fine now.

**Comment by ltcyx** [comment]
This is apparently already in the pipeline as https://github.com/Bukkit/CraftBukkit/pull/1053
